boa/src/builtins/array/mod.rs

```diff
@@ -960,7 +960,8 @@ impl Array {
     /// [spec]: https://tc39.es/ecma262/#sec-array.prototype.reduce
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce
     pub(crate) fn reduce(this: &Value, args: &[Value], interpreter: &mut Context) -> Result<Value> {
-        let this = this.to_object(interpreter)?;
+        let obj_this = this.to_object(interpreter)?;
+        let this = Value::Object(obj_this);
```

Comment:
```suggestion
        let this = this.to_object(interpreter)?.into();
```
we can `Into<Value>` for this.